backend/remote/backend_common.go

```diff
@@ -227,6 +228,59 @@ func (b *Remote) parseVariableValues(op *backend.Operation) (terraform.InputValu
 	return result, diags
 }
 
+func (b *Remote) costEstimate(stopCtx, cancelCtx context.Context, op *backend.Operation, r *tfe.Run) error {
+	if r.CostEstimate == nil {
+		return nil
+	}
+
+	if b.CLI != nil {
+		b.CLI.Output("\n------------------------------------------------------------------------\n")
+	}
+
+	// Retrieve the cost estimation to get its current status.
+	ce, err := b.client.CostEstimates.Read(stopCtx, r.CostEstimate.ID)
+	if err != nil {
+		return generalError("Failed to retrieve cost estimation", err)
+	}
+
+	msgPrefix := "Cost estimation"
+	if b.CLI != nil {
+		b.CLI.Output(b.Colorize().Color(msgPrefix + ":\n"))
+	}
+
+	// FIXME: we need to loop for this instead
```

Comment:
FIXME here. The run gets stuck on this. We need to poll for one of the statuses below.

Reply:
```Error: Failed to retrieve cost estimation: The struct field was not of a known number type

The configured "remote" backend encountered an unexpected error. Sometimes
this is caused by network connection problems, in which case you could retry
the command. If the issue persists please open a support ticket to get help
resolving the problem.```